std/algebra/twistededwards/bandersnatch/point.go

```diff
@@ -98,41 +107,87 @@ func (p *Point) Double(api frontend.API, p1 *Point, curve EdCurve) *Point {
 	return p
 }
 
+// phi endomorphism √-2 ∈ 𝒪₋₈
+// (x,y) → λ × (x,y) s.t. λ² = -2 mod Order
+func (p *Point) phi(api frontend.API, p1 *Point, curve EdCurve) *Point {
+
+	xy := api.Mul(p1.X, p1.Y)
+	yy := api.Mul(p1.Y, p1.Y)
+	f := api.Sub(1, yy)
+	f = api.Mul(f, curve.endo1)
+	g := api.Add(yy, curve.endo0)
+	g = api.Mul(g, curve.endo0)
+	h := api.Sub(yy, curve.endo0)
+
+	p.X = api.DivUnchecked(f, xy)
+	p.Y = api.DivUnchecked(g, h)
+
+	return p
+}
+
+var scalarDecompositionHint = hint.NewStaticHint(func(curve ecc.ID, inputs []*big.Int, res []*big.Int) error {
+	// TODO: handle properly negative scalars
+	var lambda, order big.Int
```

Comment:
lambda, order and glvBasis shouldn't be initialized at each hint call --> if a circuit has multiple call to this hint, the overhead on perf will be significant in the solver. 

One way to do that would be; 
create a global variable (a new struct) with the 3 elements. 

Inside the hint, use `sync.Once` to do the init (SetString + PrecomputeLattice) only once 👍

Reply:
As a side-note, this is the approach taken in WIP branch which merges different short Weierstrass implementations ([here](https://github.com/ConsenSys/gnark/pull/238/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-44b01c5b58d0588ce59dc5c74b729d43e99a85326fc34f919178f90b799bee2eR55)).

It would be great to follow similar approach, then it would be easier later to unify the implementations.